beets/dbcore/query.py

```diff
@@ -428,6 +416,24 @@ def col_clause(self) -> Tuple[str, Sequence[SQLiteType]]:
                 return "1", ()
 
 
+class InQuery(Generic[AnySQLiteType], FieldQuery[Sequence[AnySQLiteType]]):
+    """Query which matches values in the given set."""
+
+    field: str
+    pattern: Sequence[AnySQLiteType]
+    fast: bool = True
+
+    def col_clause(self) -> Tuple[str, Sequence[SQLiteType]]:
```

Comment:
```suggestion
    def col_clause(self) -> Tuple[str, Sequence[AnySQLiteType]]:
```
This returns `self.pattern` which is `AnySQLiteType`, which is more specific than `SQLiteType`.

Reply:
I wasn't sure how to type the return type here, especially now that we're mapping the initial `bytes` sequence in `pattern` to a sequence of `memoryview`.

If I use `AnySQLiteType`, then it breaks since `memoryview` != `bytes`. Ideally, I'd want to define a type variable to make sure the return type from `subvals` is the same as here

```python
    if not isinstance(path, str):
    @property
    def subvals(self) -> Sequence[T]:
        return self.pattern

    def col_clause(self) -> Tuple[str, Sequence[T]]:
        placeholders = ", ".join(["?"] * len(self.subvals))
        return f"{self.field} IN ({placeholders})", self.subvals
```

But then trying to use a type variable here gave me this mess 😁 
![image](https://github.com/beetbox/beets/assets/16212750/80dd5c70-b4c9-40c7-bb85-d624d1604382)

Reply:
Note that the interface defines this as `SQLiteType`

```python
class FieldQuery(Query, Generic[P]):
    """An abstract query that searches in a specific field for a
    pattern. Subclasses must provide a `value_match` class method, which
    determines whether a certain pattern string matches a certain value
    string. Subclasses may also provide `col_clause` to implement the
    same matching functionality in SQLite.
    """
    ...
    def col_clause(self) -> Tuple[Optional[str], Sequence[SQLiteType]]:
        return None, ()
```

Reply:
Ok, looks like I didn't check the `FieldQuery` interface properly myself. As you noted, it is conceivable that `pattern` and `col_clause` have distinct types. Let's keep it as is, if there's really the potential to do so, we can always tighten up our typing later on.